apollo-ios/Sources/ApolloSQLite/SQLiteDotSwiftDatabase.swift

```diff
@@ -10,7 +10,26 @@ public final class SQLiteDotSwiftDatabase: SQLiteDatabase {
   private let records: Table
   private let keyColumn: SQLite.Expression<CacheKey>
   private let recordColumn: SQLite.Expression<String>
-  
+
+  public enum JournalMode: String {
+    /// The rollback journal is deleted at the conclusion of each transaction. This is the default behaviour.
+    case delete = "DELETE"
+    /// Commits transactions by truncating the rollback journal to zero-length instead of deleting it.
+    case truncate = "TRUNCATE"
+    /// Prevents the rollback journal from being deleted at the end of each transaction. Instead, the header
+    /// of the journal is overwritten with zeros.
+    case persist = "PERSIST"
+    /// Stores the rollback journal in volatile RAM. This saves disk I/O but at the expense of database
+    /// safety and integrity.
+    case memory = "MEMORY"
+    /// Uses a write-ahead log instead of a rollback journal to implement transactions. The WAL journaling
+    /// mode is persistent; after being set it stays in effect across multiple database connections and after 
+    /// closing and reopening the database.
+    case wal = "WAL"
+    /// Disables the rollback journal completely
+    case off = "OFF"
+  }
```

Comment:
I suppose there might be but these come directly from [sqlite.org documentation](https://www.sqlite.org/pragma.html) so if there is an sqlite implementation that differs enough you'll probably want to write your own `SQLiteDatabase` implementation too at which point you get to define the new set of journal modes.

As for the generic parameter I simply wanted something that was able to be expressible as a `String` since that's how the pragma statement gets built. Moving the enum into `SQLiteDatabase` will serve the vast majority of users although I didn't like how protocols can't hold enums so we lost some namespacing there.

Reply:
Refactored as we discussed in https://github.com/apollographql/apollo-ios-dev/pull/443/commits/27cc39904cbfa1fc13559906656108da503e594c.